agent/services/task_service.py

```
@@ -206,6 +206,89 @@ def get_active_tasks(self) -> List[TaskEvent]:
+        FINAL_STATES = {
+            'task-succeeded', 'task-failed', 'task-revoked',
+            'task-rejected', 'task-retried',
+        }
+
+        # Latest orphaned event per task (same subquery as before, now in service)
+        latest_orphaned_sq = (
+            self.session.query(
+                TaskEventDB.task_id,
+                func.max(TaskEventDB.timestamp).label('max_timestamp'),
+            )
+            .filter(TaskEventDB.is_orphan == True)
```

Ruff (0.15.10)

[error] Avoid equality comparisons to `True`; use `TaskEventDB.is_orphan:` for truth checks

Replace with `TaskEventDB.is_orphan`

(E712)